src/NHibernate/Bytecode/EmitUtil.cs

```diff
@@ -70,29 +13,25 @@ public static void EmitBoxIfNeeded(ILGenerator il, System.Type type)
 			}
 		}
 
-		private static Dictionary<System.Type, OpCode> typeToOpcode;
+		private static readonly Dictionary<System.Type, OpCode> typeToOpcode;
 
 		static EmitUtil()
 		{
-			typeToOpcode = new Dictionary<System.Type, OpCode>(12);
-
-			typeToOpcode[typeof(bool)] = OpCodes.Ldind_I1;
-			typeToOpcode[typeof(sbyte)] = OpCodes.Ldind_I1;
-			typeToOpcode[typeof(byte)] = OpCodes.Ldind_U1;
-
-			typeToOpcode[typeof(char)] = OpCodes.Ldind_U2;
-			typeToOpcode[typeof(short)] = OpCodes.Ldind_I2;
-			typeToOpcode[typeof(ushort)] = OpCodes.Ldind_U2;
-
-			typeToOpcode[typeof(int)] = OpCodes.Ldind_I4;
-			typeToOpcode[typeof(uint)] = OpCodes.Ldind_U4;
-
-			typeToOpcode[typeof(long)] = OpCodes.Ldind_I8;
-			typeToOpcode[typeof(ulong)] = OpCodes.Ldind_I8;
-
-			typeToOpcode[typeof(float)] = OpCodes.Ldind_R4;
-
-			typeToOpcode[typeof(double)] = OpCodes.Ldind_R8;
+			typeToOpcode = new Dictionary<System.Type, OpCode>(12)
+			{
+				[typeof(bool)] = OpCodes.Ldind_I1,
```

Comment:
Isn't that what this is?  This is what ReSharper suggested replacing the Dictionary initialization with.